test/suites/greenboot.robot

```diff
@@ -0,0 +1,135 @@
+    Greenboot Healthy
+    Add User Workload
+    Microshift Cleanup And Start
+
+    [Teardown]    Run Keywords
+    ...    Cleanup User Config
+    ...    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Service Failure
+    [Documentation]    Microshift Service failure
+    Greenboot Healthy
+    Microshift Disrupt Service
+    Cleanup MicroShift    "--keep-images"
+    Run Keyword And Expect Error    	0 != 1
+    ...    Systemctl    enable    --now microshift
+    Run Keyword And Expect Error    	0 != 1
+    ...    Greenboot Healthy
+
+    [Teardown]    Run Keywords
+    ...    Microshift Restore Service
+    ...    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Pod Failure
+    [Documentation]    Microshift Pod Network failure
+    Greenboot Healthy
+    Microshift Disrupt Pod Network
+    Restart MicroShift
+    Run Keyword And Expect Error    	0 != 1
+    ...    Greenboot Healthy
+
+    [Teardown]    Run Keywords
+    ...    Clear MicroShift Config
+    ...    Microshift Cleanup And Start
+
+
+*** Keywords ***
+Setup Suite
+    [Documentation]    Set up all of the tests in this suite
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig
+    Upload String To File   ${GREENBOOT_CONFIG_CONTENT}    ${GREENBOOT_CONFIG_FILE} # change the default timeout
+
+Add User Workload
+    [Documentation]    Upload a new configuration file to the MicroShift host
+    Put File    ${LOCAL_WORKLOAD_FILE}    /tmp/busybox_running_check.sh
+    ${stdout}    ${rc}=    Execute Command
+    ...    [ -f ${GREENBOOT_WORKLOAD_FILE} ] || sudo ln -s /tmp/busybox_running_check.sh ${GREENBOOT_WORKLOAD_FILE} 
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+    Put Directory    ${MANIFEST_SOURCE_DIR}     /tmp/manifests
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp -f /tmp/manifests/* ${MANIFEST_DIR} && sudo rm -rf /tmp/manifests
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Cleanup User Config
+    ${stdout}    ${rc}=    Execute Command
+    ...    rm -rf ${GREENBOOT_WORKLOAD_FILE} ${MANIFEST_DIR}/*
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Greenboot Healthy
+    [Documentation]    Restart the greenboot-healthcheck service and check its status
+    Systemctl    restart  greenboot-healthcheck.service
+    Wait Until Greenboot Health Check Exited
+
+Microshift Disrupt Service
+    ${is_ostree}=    Check If OStree
+
+    ${cmd}=    Set Variable     sudo chmod 000 $(which hostname)
+    IF     ${is_ostree}
+        ${cmd}=    Set Variable    sudo rpm-ostree usroverlay && sudo chmod 000 $(which hostname)
+    END
+
+    ${stdout}    ${rc}=    Execute Command
+    ...    ${cmd}
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+    
+Microshift Restore Service
+    ${stdout}    ${rc}=    Execute Command
+    ...    sudo chmod 755 /usr/bin/hostname
```

Comment:
The `which` command does not work here, so can we save this in a variable and use here and in other places?